### PR TITLE
Error handler

### PR DESCRIPTION
**Pull request by kkardatos**

### CODE REVIEW

File: front-end-vue/src/main.ts [hunk 1]
```diff
@@ -145,29 +146,18 @@ const app = createApp(App)
...
-      throw new APIError(error.response.data.status, error.response.data.error, `Call to ${error.response.data.path} failed.`);
-    }
-    // 499 -> error code for request cancelled by user
-    if (error.message?.status !== 499) {
-      throw new Error(error);
-    }
+    vm.$toast.add({
+      severity: "error",
+      summary: "Request error",
+      detail: "Request for " + error.config.url.substring(error.config.url.lastIndexOf("/") + 1) + " was unsuccessful. " + error.message + ".",
```
Comment (matt-voror) [inline front-end-vue/src/main.ts:159]: Does error always have .config.url? If not this needs checked and alternate error detail provided

File: front-end-vue/src/services/CatalogueService.ts [hunk 2]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
   public static async getSearchResult(request: string, typesIris: string[]): Promise<any[]> {
-    const axiosReturn = (
-      await axios.get(this.api + "instance/search", {
+    try {
+      const axiosReturn = (await axios.get(this.api + "instance/search", {
         params: { request: request, typesIris: typesIris.join(",") }
-      })
-    ).data;
-    return axiosReturn ? axiosReturn : [];
+      })) as [];
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:10]: as any[]

File: front-end-vue/src/services/CatalogueService.ts [hunk 3]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
+      return axiosReturn || [];
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:11]: [] as any[]

File: front-end-vue/src/services/CatalogueService.ts [hunk 4]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
-    const axiosReturn = (
-      await axios.get(this.api + "instance/partial", {
+    try {
+      const axiosReturn = await axios.get(this.api + "instance/partial", {
         params: { iri: iri, predicate: predicates }
-      })
-    ).data;
-    return axiosReturn ? axiosReturn : {};
+      });
+      return axiosReturn || {};
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:22]: {} as any

File: front-end-vue/src/services/CatalogueService.ts [hunk 5]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
+    } catch (error) {
+      return {};
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:24]: {} as any

File: front-end-vue/src/services/CatalogueService.ts [hunk 6]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
+    }
   }
 
   public static async getTypesCount(): Promise<any[]> {
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:28]: Does this have a return type?

File: front-end-vue/src/services/CatalogueService.ts [hunk 7]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
-    const axiosReturn = (await axios.get(this.api + "instance/typesCount")).data;
-    return axiosReturn ? axiosReturn : [];
+    try {
+      const axiosReturn = (await axios.get(this.api + "instance/typesCount")) as [];
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:30]: Don't need typing here as data return is typed and should error if axios returns incorrect type. Also as [] means as nothing[]

File: front-end-vue/src/services/CatalogueService.ts [hunk 8]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
+      return axiosReturn || [];
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:31]: But do need it here for empty array to keep types correct

File: front-end-vue/src/services/CatalogueService.ts [hunk 9]
```diff
@@ -4,25 +4,33 @@ export default class CatalogueService {
...
+    } catch (error) {
+      return [];
```
Comment (matt-voror) [inline front-end-vue/src/services/CatalogueService.ts:33]: and here

File: front-end-vue/src/services/ConfigService.ts [hunk 10]
```diff
@@ -6,18 +6,24 @@ export default class ConfigService {
...
-      await axios.get(this.api + "api/config/componentLayout", {
+    try {
+      const axiosReturn = (await axios.get(this.api + "api/config/componentLayout", {
         params: {
           name: name
         }
-      })
-    ).data;
-    return axiosReturn ? axiosReturn : ([] as DefinitionConfig[]);
+      })) as DefinitionConfig[];
```
Comment (matt-voror) [inline front-end-vue/src/services/ConfigService.ts:14]: see prev

File: front-end-vue/src/services/ConfigService.ts [hunk 11]
```diff
@@ -6,18 +6,24 @@ export default class ConfigService {
...
+    } catch (error) {
+      return [] as DefinitionConfig[];
+    }
   }
 
   public static async getFilterDefaults(): Promise<FilterDefaultsConfig> {
-    const axiosReturn = (await axios.get(this.api + "api/config/filterDefaults")).data;
-    return axiosReturn ? axiosReturn : ({} as FilterDefaultsConfig);
+    try {
+      const axiosReturn = (await axios.get(this.api + "api/config/filterDefaults")) as FilterDefaultsConfig;
```
Comment (matt-voror) [inline front-end-vue/src/services/ConfigService.ts:23]: see prev

File: front-end-vue/src/services/EntityService.ts [hunk 12]
```diff
@@ -17,195 +17,225 @@ export default class EntityService {
...
+      const axiosReturn = await axios.get(this.api + "api/entity/partial", {
         params: {
           iri: iri,
           predicate: predicates.join(",")
         }
-      })
-    ).data;
-    return axiosReturn ? axiosReturn : {};
+      });
+      return axiosReturn || {};
```
Comment (matt-voror) [inline front-end-vue/src/services/EntityService.ts:27]: {} as any

File: front-end-vue/src/services/EntityService.ts [hunk 13]
```diff
@@ -17,195 +17,225 @@ export default class EntityService {
...
+    } catch (error) {
+      return {};
```
Comment (matt-voror) [inline front-end-vue/src/services/EntityService.ts:29]: {} as any

File: front-end-vue/src/services/EntityService.ts [hunk 14]
```diff
@@ -17,195 +17,225 @@ export default class EntityService {
...
+    try {
+      const axiosReturn = (await axios.get(this.api + "api/entity/partialBundle", {
         params: {
           iri: iri,
           predicate: predicates.join(",")
         }
-      })
-    ).data;
-    return axiosReturn ? axiosReturn : ({} as PartialBundle);
+      })) as PartialBundle;
```
Comment (matt-voror) [inline front-end-vue/src/services/EntityService.ts:40]: see prev and apply to all below

**Review by matt-voror — CHANGES_REQUESTED** [review]

File: front-end-vue/src/main.ts (hunk 1 above)
Comment (kkardatos) [reply front-end-vue/src/main.ts:159]: since they are all axios error interceptions, config.url should always be present

**Pull request merged**